### PR TITLE
Update docs requirements

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:** Sphinx released a new version recently, we need to pin some extensions to keep it working with our pinned Sphinx version.

**Description of the Change:** Pinning Sphinx extensions.

**Benefits:** Docs will build

**Related PennyLane PRs:** https://github.com/PennyLaneAI/pennylane/pull/5062 https://github.com/PennyLaneAI/catalyst/pull/460

### CODE REVIEW

**Review by multiphaseCFD — APPROVED** [review]
Looks good to me! Thanks @AmintorDusko.

File: doc/requirements.txt [hunk 1]
```diff
@@ -5,6 +5,11 @@ graphviz
 pybind11
 sphinx
 sphinx-automodapi
```
Comment (multiphaseCFD) [inline doc/requirements.txt:7]: Just a minor question. Do we want to pin sphinx to be `5.0`+? The error found in my readthedocs CI is `sphinx.errors.VersionRequirementError: 5.`0

**Review by vincentmr — APPROVED** [review]
Looks good to me, cheers @AmintorDusko .

File: doc/requirements.txt (hunk 1 above)
Comment (AmintorDusko) [reply doc/requirements.txt:7]: No, we don't. At least not for now.

**Pull request merged**